apps/examples/st_things/sample_light_things.c

```diff
@@ -0,0 +1,145 @@
+/* ****************************************************************
+ *
+ * Copyright 2017 Samsung Electronics All Rights Reserved.
+ *
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************/
+
+#include <tinyara/config.h>
+
+#include <stdio.h>
+#include <string.h>
+#include <stdbool.h>
+#include <signal.h>
+#include "st_things/st_things.h"
+#include <iotbus/iotbus_gpio.h>
+
+#define TAG                     "LIGHT_THINGS"
+#define SW703 44
+
+char *g_resource_uri;
+
+static const char *RES_SWITCH = "/switch";
+
+extern bool handle_get_request_on_switch(st_things_get_request_message_s * req_msg, st_things_representation_s * resp_rep);
+extern bool handle_set_request_on_switch(st_things_set_request_message_s * req_msg, st_things_representation_s * resp_rep);
+
+void gpio_callback_event(void *user_data)
+{
+	printf("gpio_callback_event!!\n");
+	printf("reset :: %d\n", st_things_reset());
+}
+
+void handle_reset_result(bool result)
+{
+	printf("[%s]Reset %s.\n", result ? "succeeded" : "failed", TAG);
+}
+
+void handle_things_status_change(st_things_status_e things_status)
+{
+	printf("[%s]Things status is changed: %d\n", TAG, things_status);
+}
+
+int g_quit_flag = 0;
+
+void handle_main_loop(void)
+{
+	while (!g_quit_flag) {
```

Comment:
There is no code to set g_quit_flag to 1. Is this infinite loop? Then why don't you use while (1)?

Reply:
@KimDongEon Please review

Reply:
g_quit_flag is an unused variable.
It was supposed to be used as a parameter for ending future easysetup.

So I added a description instead of deleting the variable.